invoker/buildpack.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2019 The original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package invoker
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/buildpack/libbuildpack/buildplan"
+	"github.com/cloudfoundry/libcfbuildpack/build"
+	"github.com/cloudfoundry/libcfbuildpack/detect"
+	"github.com/projectriff/riff-buildpack/metadata"
+)
+
+const (
+	Error_Initialize          = 101
+	Error_ReadMetadata        = 102
+	Error_DetectedNone        = 103
+	Error_DetectAmbiguity     = 104
+	Error_UnsupportedLanguage = 105
+	Error_DetectInternalError = 106
+)
+
+type Buildpack interface {
+	Name() string
+	Detect(detect detect.Detect, metadata metadata.Metadata) (bool, error)
+	BuildPlan(detect detect.Detect, metadata metadata.Metadata) buildplan.BuildPlan
+	Invoker(build build.Build) (Invoker, bool, error)
+}
+
+type Invoker interface {
+	Contribute() error
+}
+
+type BuildpackCommands struct {
+	buildpack Buildpack
+}
+
+func NewBuildpackCommands(buildpack Buildpack) *BuildpackCommands {
+	return &BuildpackCommands{
+		buildpack: buildpack,
+	}
+}
+
+func (bc *BuildpackCommands) Name() string {
+	return bc.buildpack.Name()
+}
+
+func (bc *BuildpackCommands) Detect() {
+	detect, err := detect.DefaultDetect()
+	if err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Failed to initialize Detect: %s\n", err)
+		os.Exit(Error_Initialize)
+	}
+
+	if err := detect.BuildPlan.Init(); err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "Failed to initialize Build Plan: %s\n", err)
+		os.Exit(Error_Initialize)
+	}
+
+	if code, err := bc.detect(detect); err != nil {
+		detect.Logger.Info(err.Error())
```

Comment:
Should use something else than Info (or at least use visual cue that we're dealing with an error)

Reply:
The logger only exposes Debug and Info levels. It appears the newer version of libcfbuildpack support more logging levels.